src/parameterizations/lateral/MOM_mixed_layer_restrat.F90

```diff
@@ -928,14 +937,20 @@ subroutine mixedlayer_restrat_Bodner(CS, G, GV, US, h, uhtr, vhtr, tv, forces, d
     CS%MLD_filtered_slow(i,j) = big_H(i,j)
   enddo ; enddo
 
+  if (allocated(tv%SpV_avg) .and. .not.(GV%Boussinesq .or. GV%semi_Boussinesq)) then
+    uflux_rescale = US%Z_to_L * GV%RZ_to_H / tv%SpV_avg(i,j,1)
```

Comment:
In non-Boussinesq mode, uflux_rescale is a function of i and j, and it therefore either needs to be turned into a 2-d array or calculated inside of the relevant i- and j- loops.

Reply:
:facepalm: